### PR TITLE
Force cozy bar to rerender when locale change

### PR DESCRIPTION
**Pull request by CPatchane**
Expose setLocale to change the cozy bar locale from the application

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -46,6 +46,11 @@ const bindEvents = function CozyBarBindEvents () {
   this._clickOutsideListener = () => this.fire('clickOutside')
   body.addEventListener('click', this._clickOutsideListener)
 
+  document.getElementById('coz-bar').addEventListener('localechange', () => {
+    console.log(this)
```
Comment (gregorylegarec) [inline src/index.js]: ⚠️  `console.log`

File: src/index.js [hunk 2]
```diff
@@ -138,12 +145,19 @@ const init = function CozyBarInit ({
...
 
   if (view) {
     bindEvents.call(view)
     view.on('teardown', unbindEvents.bind(view))
   }
 }
 
-module.exports = { init, version: __VERSION__ }
+// set the cozy bar locale from the application
+const setLocale = function SetLocale (lang) {
```
Comment (gregorylegarec) [inline src/index.js]: Is `SetLocale` mandatory between `function` and `(lang)` ?

File: src/lib/i18n.js [hunk 3]
```diff
@@ -20,8 +20,18 @@ const init = function I18nInit (lang) {
   }
 }
 
+const setLang = function I18nSet (lang) {
```
Comment (gregorylegarec) [inline src/lib/i18n.js]: Why `setLang` and not `setLocale` ?

File: src/index.js (hunk 1 above)
Comment (CPatchane) [inline src/index.js]: Oops! It's fixed^^

File: src/lib/i18n.js (hunk 3 above)
Comment (CPatchane) [inline src/lib/i18n.js]: It was due to the `setLocale` already used in `index.js`, but I renamed it `i18nSetLocale` for better understanding here.

File: src/index.js (hunk 2 above)
Comment (CPatchane) [inline src/index.js]: I really don't know for now but all functions and methods are declared in this way in the cozy bar code so I just followed to be coherent.

File: src/components/Bar.svelte [hunk 4]
```diff
@@ -4,7 +4,7 @@
 </button>
 {{/if}}
 
-<h1 class='{{titleClass}}'>
+<h1 data-lang='{{lang}}' class='{{titleClass}}'>
```
Comment (lespacedunmatin) [inline src/components/Bar.svelte]: Why `data-lang` ? You can also use `lang` directly (is useful for screen readers).

File: src/index.js (hunk 2 above)
Comment (gregorylegarec) [inline src/index.js]: So, let's be coherent and keep it as it for now.

File: src/components/Bar.svelte (hunk 4 above)
Comment (CPatchane) [inline src/components/Bar.svelte]: In fact I added this attribute just to use the lang data property in the component template to get the correct rerendering behaviour when the lang change. I don't have opinions about that, just I know data-lang is also accepted as attribute. So let's do it 👍

**Comment by gregorylegarec** [comment]
Thanks @CPatchane

File: src/index.js (hunk 2 above)
Comment (m4dz) [inline src/index.js]: Having functions stored in variables allow JS engine to evaluate them lazily, which is better for perfomances, but naming functions even in this case make debug messages more explicits :smiley:

**Pull request merged**